elytron/WFCORE-5027-charset-encoding-support.adoc

```diff
@@ -0,0 +1,240 @@
+
+=== Other Interested Projects
+
+N/A
+
+== Requirements
+
+=== Hard Requirements
+
+All realms that support hashed passwords need to be modified: https://github.com/wildfly-security/wildfly-elytron/blob/1.x/auth/realm/base/src/main/java/org/wildfly/security/auth/realm/LegacyPropertiesSecurityRealm.java[Properties Realm],
+https://github.com/wildfly-security/wildfly-elytron/blob/1.x/auth/realm/base/src/main/java/org/wildfly/security/auth/realm/FileSystemSecurityRealm.java[Filesystem Realm],
+https://github.com/wildfly-security/wildfly-elytron/blob/1.x/auth/realm/jdbc/src/main/java/org/wildfly/security/auth/realm/jdbc/JdbcSecurityRealm.java[JDBC Realm],
+and https://github.com/wildfly-security/wildfly-elytron/blob/1.x/auth/realm/ldap/src/main/java/org/wildfly/security/auth/realm/ldap/LdapSecurityRealm.java[LDAP Realm].
+Note the Filesystem realm has been added to the scope of this RFE even if it was not available in the PicketBox configuration,
+as the properties realm is a legacy configuration.
+
+
+Two new attributes will be added to all of these realms: ``hash-encoding`` and
+``hash-charset``.
+
+
+* ``hash-encoding``: This attribute specifies the string format for the hashed password if a hashing algorithm is specified.
+It may specify one of three: ``hex``, ``base64``, and ``rcf2617``. It will be set to ``base-64`` by default if omitted
+to replicate the legacy configuration options.
+
+* ``hash-charset``: This attribute specifies the name of the character set to use when converting the password string to
+a byte array. It will be set to the default encoding set in the container's runtime environment by default if omitted
+to replicate the legacy configuration options.
+
+Though this is the general definition for what these attributes mean, they will be configured in
+different ways depending on the security realm. See below for the realm-specific configurations.
+
+
+==== Properties Realm
+
+The ``hash-encoding`` and ``hash-charset`` attributes will be configured as resource-wide
+attributes, as all the passwords within a properties file follow the same format and
+are either clear text or already digested. For example:
+
+    /subsystem=elytron/properties-realm=propRealm:add(groups-attribute=groups, groups-properties={...}, users-properties={...), \
+    plain-text=false, hash-encoding="hex", hash-charset="ISO-8859-5")
+
+
+
+The configuration for the properties realm is similar to the ``UserRolesLoginModule`` in the PicketBox configuration
+since these options are resource-wide i.e. these hashing options will be
+used when verifying every single client-provided password against the credentials stored in the
+security realm, as long as hashing is enabled. The only difference lies in that the Elytron Properties realm
+only allows ClearPasswords or DigestPasswords, as opposed to the ``UserRolesLoginModule`` which supports a wider variety of
+hashing algorithms. Supporting more hashing algorithms is out of the scope of this RFE.
```

Comment:
@fjuma For now, I have written that supporting all the hashing algorithms that legacy supported is out of the scope of this RFE. Should it be included?